RecoTracker/TkNavigation/plugins/SimpleNavigationSchool.cc

```diff
@@ -126,6 +126,13 @@ void SimpleNavigationSchool::linkNextForwardLayer( const BarrelDetLayer* bl,
 	fli != theRightLayers.end(); fli++) {
     if ( length < (**fli).position().z() &&
 	 radius < (**fli).specificSurface().outerRadius()) {
+      //search if there are any sovrapposition between forward layers
+      for ( FDLI fliNext = fli; fliNext != theRightLayers.end(); fliNext++) {
+        if ( (**fliNext).position().z() < (**fli).position().z() && (**fliNext).specificSurface().innerRadius() < (**fli).specificSurface().outerRadius()) {
```

Comment:
Yes, it is because they are `theRightLayers` which means Forward Layers with z > 0
But maybe we can put en exception also in this case, if you prefer.